packages/ledger-icrc/src/index.canister.ts

```diff
@@ -10,6 +10,9 @@ import { IndexError } from "./errors/index.errors";
 import type { IcrcLedgerCanisterOptions } from "./types/canister.options";
 import type { GetAccountTransactionsParams } from "./types/index.params";
 
+/**
+ * @deprecated Replaced by `IcrcIndexNgCanister`, which requires interacting with an Index canister that has been upgraded to the so-called `index-ng` WASM.
```

Comment:
Do we want that?

The problem is not the library, but the canister that is deprecated. If a user interacts with an non ng-index, the user can't decide to use `IcrcIndexNgCanister`.

The user can't do anything about it if the canister is an index and not index-ng. I believe that we should set something as deprecated if there is an alternative. But in this case, there is no alternative. Users can't use IcrcIndexNgCanister.

Reply:
> The problem is not the library, but the canister that is deprecated. If a user interacts with an non ng-index, the user can't decide to use IcrcIndexNgCanister.

That's an excellent point. My thinking was more along the lines that if a developer is not the owner of a canister, they could contact the controllers to request a migration.

For example, if you're interacting with the OpenChat index canister and notice the deprecation isn't working, you could contact OC to ask their DAO to upgrade the canister.

In other words, I was thinking that setting it as deprecated could help spread the word about the upgrade and also communicate that, ultimately, this module should be removed, at least in my view.

However, you're right; we might end up never removing it.

I'm also looking at this from the perspective of Oisy, where we won't support—or at least I will strongly advocate against supporting—the "old-index" canister.

But again, that's a valid point. I'm fine with not deprecating it, just wanted to provide a bit more context.